### PR TITLE
Better repr for distributed workers.

### PR DESCRIPTION
**Pull request by cowlicks**
```python
In [1]: from dask.distributed import Scheduler, Worker; s = Scheduler(); w = Worker(s.address_to_workers); w
Out[1]: <dask.distributed.Worker at 0x7fdd4c7c26a0 address=camargo:59318, scheduler=camargo:42964>
```

### CODE REVIEW

File: dask/distributed/worker.py [hunk 1]
```diff
@@ -530,6 +530,11 @@ def close(self):
     def __del__(self):
         self.close()
 
+    def __repr__(self):
+        s = '<dask.distributed.Worker at %s address=%s, scheduler=%s>'
+        return s % (hex(id(self)), self.address.decode('utf-8')[6:],
+                    self.scheduler.decode('utf-8')[6:])
```
Comment (mrocklin) [inline dask/distributed/worker.py]: We can probably leave off the hex(id(...)) bit and probably the `<dask.distributed` bit.  They don't provide much information and once we have the two addresses we've probably completely identified the worker, removing the need for the `id`.

Comment (cowlicks) [inline dask/distributed/worker.py]: 2 workers on the same machine can share the same `address` (`port_to_workers`) so it doesn't uniquely identify them.

Comment (mrocklin) [inline dask/distributed/worker.py]: Wouldn't two workers with the same address/port conflict on port?

Comment (cowlicks) [inline dask/distributed/worker.py]: Oops, yes you are correct. I was misremembering a conversation we had about this and adding one worker per processor on anaconda cluster.

Comment (cowlicks) [inline dask/distributed/worker.py]: Great, I'm glad we can remove the `id` it is too verbose as is.

**Comment by cowlicks** [comment]
```python
In [1]: from dask.distributed import Worker, Scheduler

In [2]: s = Scheduler(); w = Worker(s.address_to_workers)

In [3]: w
Out[3]: <dask.distributed.Worker address=camargo:56956, scheduler=camargo:55944>
```

**Pull request merged**